### PR TITLE
Add docs versioning and release notes

### PR DESCRIPTION
**Pull request by mgoulao**
# Description

* Add workflows to build docs when a new version is released
* Add a Release Notes page to docs

### CODE REVIEW

**Comment by LucasAlegre** [comment]
Hey @mgoulao , thanks for the PR! Is this ready for merging?

**Comment by mgoulao** [comment]
Yes.

File: docs/conf.py [hunk 1]
```diff
@@ -23,7 +24,9 @@
 author = "Farama Foundation"
 
 # The full version, including alpha/beta/rc tags
-release = "0.3.0"
+# TODO: uncomment when the attr __version__ is added
+# release = mo_gymnasium.__version__
```
Comment (ffelten) [inline docs/conf.py:28]: Where should we add this attribute?

Comment (LucasAlegre) [reply docs/conf.py]: Just did

**Review by ffelten — APPROVED** [review]

**Pull request merged**